media-proxy/src/mesh/conn_rdma_rx.cc

```diff
@@ -142,14 +142,39 @@ void RdmaRx::rdma_cq_thread(context::Context& ctx) {
         } else if (ret == -EAGAIN) {
             // No events to process, yield CPU briefly to avoid busy looping
             std::this_thread::sleep_for(std::chrono::microseconds(CQ_RETRY_DELAY_US));
+        } else if (ret == -FI_EAVAIL) {
+            // Read the error details
+            struct fi_cq_err_entry err_entry;
+            int err_ret = fi_cq_readerr(ep_ctx->cq_ctx.cq, &err_entry, 0);
+            if (err_ret >= 0) {
+                if (err_entry.err == FI_ECONNRESET || err_entry.err == -107) {
+                    log::warn("RDMA connection reset or endpoint not connected. Waiting for new "
+                              "connection.")("error", fi_strerror(err_entry.err))
+                              ("kind",kind2str(_kind));
+                    thread::Sleep(ctx, std::chrono::milliseconds(1000)); // Pause before retrying
+                } else {
+                    log::error("RDMA rx encountered an error in CQ")(
+                        "error", fi_strerror(err_entry.err))("kind", kind2str(_kind));
+                }
+            } else {
+                log::error("RDMA rx failed to read CQ error entry")
+                          ("error", fi_strerror(-err_ret))("kind", kind2str(_kind));
+            }
+        } else if (ret == -107) {
```

Comment:
Can this `-107` be changed to some predefined const of the underlying lib that you use?

Reply:
Fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/293/commits/907a4e3917a16fa35b4ebde105a15f179531b040